### PR TITLE
Use gh to paginate file listing

### PR DESCRIPTION
**Pull request by hluotola-paf**

### CODE REVIEW

File: terragrunt/find-changes/action.yml [hunk 1]
```diff
@@ -19,37 +16,30 @@ outputs:
...
+          pull_request)
+            changed_files="$(gh api repos/{owner}/{repo}/pulls/${PR_NUMBER}/files --paginate)"
+            ;;
+
+          push)
+            changed_files="$(gh api repos/{owner}/{repo}/commits/${HEAD_COMMIT} --paginate --jq .files)"
+            ;;
+
+          *)
+            changed_files=""
```
Comment (jpalomaki) [inline terragrunt/find-changes/action.yml:39]: Should we just err explicitly here (in case of unsupported event)?

Comment (hluotola-paf) [reply terragrunt/find-changes/action.yml]: yep

**Review by conradt-paf — APPROVED** [review]
👍

**Review by jpalomaki — APPROVED** [review]

**Review by conradt-paf — APPROVED** [review]
👍

**Review by jpalomaki — APPROVED** [review]

File: terragrunt/find-changes/action.yml [hunk 2]
```diff
@@ -63,3 +53,7 @@ runs:
         echo ::set-output name=updates::$(echo -n ${updates} | jq -Rsc 'split(" ")')
         echo ::set-output name=deletions::$(echo -n ${deletions} | jq -Rsc 'split(" ")')
       shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.github-token }}
```
Comment (conradt-paf) [inline terragrunt/find-changes/action.yml:57]: Can that be `${{ secrets.github-token }}` ?

Comment (hluotola-paf) [reply terragrunt/find-changes/action.yml:57]: Composite actions cannot read secrets.

**Pull request merged**